vcd/resource_vcd_nsxt_alb_pool.go

```diff
@@ -0,0 +1,646 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/util"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceVcdAlbPool() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdAlbPoolCreate,
+		ReadContext:   resourceVcdAlbPoolRead,
+		UpdateContext: resourceVcdAlbPoolUpdate,
+		DeleteContext: resourceVcdAlbPoolDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdAlbPoolImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway ID in which ALB Pool should be created",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of ALB Pool",
+			},
+			"enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Boolean value if ALB Pool is enabled or not (default true)",
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of ALB Pool",
+			},
+			"algorithm": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Algorithm for choosing pool members (default LEAST_CONNECTIONS)",
```

Comment:
Maybe is worth adding possible supported values?

Reply:
I didn't want to make it lengthy. But added now.